src/main/java/io/quarkus/bot/MarkClosedPullRequestInvalid.java

```diff
@@ -23,6 +25,18 @@ void markClosedPullRequestInvalid(@PullRequest.Closed GHEventPayload.PullRequest
         GHPullRequest pullRequest = pullRequestPayload.getPullRequest();
 
         if (pullRequest.isMerged()) {
+            if (!pullRequest.getLabels().isEmpty()) {
+                if (!quarkusBotConfig.isDryRun()) {
+                    Collection<GHLabel> labels = pullRequest.getLabels();
+                    for (GHLabel label : labels) {
+                        if (label.getName().equals(Labels.TRIAGE_WAITING_FOR_CI)) {
+                            pullRequest.removeLabels(Labels.TRIAGE_WAITING_FOR_CI);
```

Comment:
You can `break;` from the for loop as soon as you removed it as you won't find a second one (labels are unique).